src/Data/Vec/Base.agda

```diff
@@ -358,3 +355,20 @@ last xs with initLast xs
 transpose : Vec (Vec A n) m → Vec (Vec A m) n
 transpose []         = replicate []
 transpose (as ∷ ass) = replicate _∷_ ⊛ as ⊛ transpose ass
+
+
+------------------------------------------------------------------------
+-- DEPRECATED
+------------------------------------------------------------------------
+-- Please use the new names as continuing support for the old names is
+-- not guaranteed.
+
+-- Version 2.0
+
+join : Vec (Vec A n) n → Vec A n
+join = _>>= id where open DiagonalBind
+
+{-# WARNING_ON_USAGE join
+"Warning: join was deprecated in v2.0
+Please use Data.Vec.Effectful.join instead."
+#-}
```

Comment:
Is this deprecation necessary?
That is, was `join` only introduced in v2.0, in which case it is harmless (is it?) to simply move it?

Reply:
Actually, it seems as though `join` was only introduced in commit `7b7230752e86ba4e4b311fb812ede7cc5818a5e8`, so we can avoid the deprecation... just as well, as once again, we are deprecating/trying to deprecate a name local to the module `DiagonalBind`, not at the top-level...